### PR TITLE
Fix RowRef behavior on unknown languages; Add RawRow and RawSubrow

### PR DESCRIPTION
**Pull request by WorkingRobot**

### CODE REVIEW

**Review by NotAdam — APPROVED** [review]

File: src/Lumina/Excel/RawSubrow.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        ReadFloat32( GetColumnOffset( columnIdx ) );
+
+    public long ReadInt64Column( int columnIdx ) =>
+        ReadInt64( GetColumnOffset( columnIdx ) );
+
+    public ulong ReadUInt64Column( int columnIdx ) =>
+        ReadUInt64( GetColumnOffset( columnIdx ) );
+
+    public bool ReadPackedBoolColumn( int columnIdx, byte bit ) =>
+        ReadPackedBool( GetColumnOffset( columnIdx ), bit );
```
Comment (Haselnussbomber) [inline src/Lumina/Excel/RawSubrow.cs:85]:
```suggestion
    public bool ReadPackedBoolColumn( int columnIdx )
    {
        var column = page.Sheet.Columns[columnIdx];
        return ReadPackedBool( column.Offset, (byte)( column.Type - ExcelColumnDataType.PackedBool0 ) );
    }
```

File: src/Lumina/Excel/RawRow.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        ReadFloat32( GetColumnOffset( columnIdx ) );
+
+    public long ReadInt64Column( int columnIdx ) =>
+        ReadInt64( GetColumnOffset( columnIdx ) );
+
+    public ulong ReadUInt64Column( int columnIdx ) =>
+        ReadUInt64( GetColumnOffset( columnIdx ) );
+
+    public bool ReadPackedBoolColumn( int columnIdx, byte bit ) =>
+        ReadPackedBool( GetColumnOffset( columnIdx ), bit );
```
Comment (Haselnussbomber) [inline src/Lumina/Excel/RawRow.cs:119]:
```suggestion
    public bool ReadPackedBoolColumn( int columnIdx )
    {
        var column = page.Sheet.Columns[columnIdx];
        return ReadPackedBool( column.Offset, (byte)( column.Type - ExcelColumnDataType.PackedBool0 ) );
    }
```

File: src/Lumina/Excel/RawSubrow.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
+using Lumina.Text.ReadOnly;
+
+namespace Lumina.Excel;
+
+public readonly struct RawSubrow( ExcelPage page, uint offset, uint row, ushort subrow ) : IExcelSubrow<RawSubrow>
+{
+    public ExcelPage Page => page;
+    public uint Offset => offset;
+    public uint RowId => row;
+    public ushort SubrowId => subrow;
```
Comment (Haselnussbomber) [inline src/Lumina/Excel/RawSubrow.cs:10]:
```suggestion
    public ushort SubrowId => subrow;

    public IReadOnlyList< ExcelColumnDefinition > Columns =>
        page.Sheet.Columns;
```

File: src/Lumina/Excel/RawSubrow.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    public ushort GetColumnOffset( int columnIdx ) =>
+        page.Sheet.GetColumnOffset( columnIdx );
```
Comment (Haselnussbomber) [inline src/Lumina/Excel/RawSubrow.cs:13]:
```suggestion
    public ushort GetColumnOffset( int columnIdx ) =>
        page.Sheet.GetColumnOffset( columnIdx );

    public object ReadColumn( int columnIdx )
    {
        var column = page.Sheet.Columns[columnIdx];
        return column.Type switch
        {
            ExcelColumnDataType.String => ReadString( column.Offset ),
            ExcelColumnDataType.Bool => ReadBool( column.Offset ),
            ExcelColumnDataType.Int8 => ReadInt8( column.Offset ),
            ExcelColumnDataType.UInt8 => ReadUInt8( column.Offset ),
            ExcelColumnDataType.Int16 => ReadInt16( column.Offset ),
            ExcelColumnDataType.UInt16 => ReadUInt16( column.Offset ),
            ExcelColumnDataType.Int32 => ReadInt32( column.Offset ),
            ExcelColumnDataType.UInt32 => ReadUInt32( column.Offset ),
            ExcelColumnDataType.Float32 => ReadFloat32( column.Offset ),
            ExcelColumnDataType.Int64 => ReadInt64( column.Offset ),
            ExcelColumnDataType.UInt64 => ReadUInt64( column.Offset ),
            >= ExcelColumnDataType.PackedBool0 and <= ExcelColumnDataType.PackedBool7 =>
                page.ReadPackedBool( column.Offset, (byte)( column.Type - ExcelColumnDataType.PackedBool0 ) ),
            _ => throw new InvalidOperationException( $"Unknown column type {column.Type}" )
        };
    }
```

**Review by Haselnussbomber — COMMENTED** [review]
I suggested these changes in the Dalamud Discord earlier today:
- RawSubrow is missing the Columns property and ReadColumn function which are present in RawRow
- ReadPackedBoolColumn shouldn't take a bit, instead it should be derived from the columns ExcelColumnDataType instead, like how ReadColumn is doing it.

**Comment by Haselnussbomber** [comment]
I PR'd the suggestions to your fork instead, because the suggestions here didn't include `using` directives:
- https://github.com/WorkingRobot/Lumina/pull/6

**Comment by Haselnussbomber** [comment]
I think the only thing still missing is a way to get rows from `RawExcelSheet` (which you get from calling `GameData.Excel.GetRawSheet()`).

**Pull request merged**